### PR TITLE
[Workspaces]Add features in use case card and preselect first use case

### PR DESCRIPTION
**Pull request by wanglam**
### Description

This PR is for adding features to workspace use case card in the workspace form and preselect the first use case in the workspace create page. They mainly includes below changes:
1. Add feature titles to the workspace use case object
2. Support display feature titles in the workspace use case card
3. Assign first available use case as default name and features to workspace form
4. Automatic update workspace name after use case changed if no manual input workspace name 

### Issues Resolved

#7705 

## Screenshot

image

## Testing the changes

- Clone branch code and run `yarn osd bootstrap --single-version ignore`
- Add below configs in `config/opensearch_dashboards.yml`
```
savedObjects.permission.enabled: true
workspace.enabled: true
uiSettings:
  overrides:
    'home:useNewHomePage': true
```
- Run `yarn start --no-base-path`
- Login and visit workspace create page
- Features will be displayed in the use case card, and it can be expand or collapse

## Changelog

- feat: [Workspaces]Add features in use case card and preselect first use case

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/workspace/public/components/workspace_form/workspace_form.tsx [hunk 1]
```diff
@@ -29,27 +29,43 @@ export const WorkspaceForm = (props: WorkspaceFormProps) => {
...
+      setName(newName);
+      nameManualChangedRef.current = true;
+    },
+    [setName]
+  );
+  const handleUseCaseChange = useCallback(
+    (newUseCase) => {
+      handleUseCaseChangeInHook(newUseCase);
+      const useCase = availableUseCases.find((item) => newUseCase === item.id);
+      if (!nameManualChangedRef.current && useCase) {
```
[the remark this answers is not in the document]
Comment (wanglam) [reply src/plugins/workspace/public/components/workspace_form/workspace_form.tsx:63]: Sure. The name only been updated when user doesn't manual change the workspace name and select another use case. For example, the page select the `observability` use case and set the workspace name to `observability` by default. The workspace name will be changed to `Security Analytics` after user select a `Security Analytics` use case.

**Review by SuZhou-Joe — APPROVED** [review]

File: src/plugins/workspace/public/components/workspace_form/workspace_form.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (SuZhou-Joe) [reply src/plugins/workspace/public/components/workspace_form/workspace_form.tsx:63]: Thanks, it was my fault that did not notice it is handleUseCaseChange.

**Review by SuZhou-Joe — DISMISSED** [review]

File: src/plugins/workspace/public/components/workspace_creator/workspace_creator.tsx [hunk 2]
```diff
@@ -116,18 +129,22 @@ export const WorkspaceCreator = (props: WorkspaceCreatorProps) => {
...
-              dataSourceManagement={dataSourceManagement}
-              availableUseCases={availableUseCases}
-              defaultValues={defaultWorkspaceFormValues}
-            />
-          )}
+          {application &&
+            savedObjects &&
+            // Default values only worked for component mount, should wait for isOnlyAllowEssential and availableUseCases loaded
+            isOnlyAllowEssential !== undefined &&
+            availableUseCases !== undefined && (
```
Comment (ruanyl) [inline src/plugins/workspace/public/components/workspace_creator/workspace_creator.tsx:136]: Maybe assign this to a readable name?
```suggestion
          {application &&
            savedObjects &&
            // Default values only worked for component mount, should wait for isOnlyAllowEssential and availableUseCases loaded
            isOnlyAllowEssential !== undefined &&
            availableUseCases !== undefined && (
```

File: src/plugins/workspace/public/components/workspace_form/use_form_available_use_cases.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      if (shouldUpdate) {
+        setIsOnlyAllowEssential(payload);
+      }
+    };
+    (async () => {
+      try {
+        const result = await getIsOnlyAllowEssentialUseCase(savedObjects.client);
+        updateEssential(result);
+      } catch (e) {
+        // Set to false is failed to fetch is only allow essential use case
```
Comment (ruanyl) [inline src/plugins/workspace/public/components/workspace_form/use_form_available_use_cases.ts:44]: This comment doesn't sound correct.

**Review by ruanyl — DISMISSED** [review]

**Review by ruanyl — APPROVED** [review]

**Pull request merged**